File: spelline-backend/src/main/java/ch/bbw/spelline/model/Request.java
package ch.bbw.spelline.model;

import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity
public class Request {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	private byte[] file;
	private int wordcount;
	private double price;
	@Enumerated(EnumType.STRING)
	private RequestLevel level;
	@Enumerated(EnumType.STRING)
	private RequestStatus status;
	@ManyToOne
	@JoinColumn(name = "userFk")
	private User user;
	@ManyToOne
	@JoinColumn(name = "expertFk")
	private User expert;

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public byte[] getFile() {
		return file;
	}

	public void setFile(byte[] file) {
		this.file = file;
	}

	public int getWordcount() {
		return wordcount;
	}

	public void setWordcount(int wordcount) {
		this.wordcount = wordcount;
	}

	public double getPrice() {
		return price;
	}

	public void setPrice(double price) {
		this.price = price;
	}

	public RequestLevel getLevel() {
		return level;
	}

	public void setLevel(RequestLevel level) {
		this.level = level;
	}

	public RequestStatus getStatus() {
		return status;
	}

	public void setStatus(RequestStatus status) {
		this.status = status;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public User getExpert() {
		return expert;
	}

	public void setExpert(User expert) {
		this.expert = expert;
	}
}

File: spelline-frontend/src/models/Request.ts
import {defaultUser, User} from './User';

export type Request = {
    id: number,
    file?: File,
    wordcount: number,
    price: number,
    level: RequestLevel,
    status: RequestStatus,
    user?: User,
    expert?: User;
};

export enum RequestLevel {
    NORMAL = "NORMAL",
    PRO = "PRO",
    EXPRESSPRO = "EXPRESSPRO"
}

export enum RequestStatus {
    OPEN = "OPEN",
    INPROGRESS= "INPROGRESS",
    DONE = "DONE"
}

export const defaultRequest: Request = {
    id: 0,
    wordcount: 0,
    price: 0,
    level: RequestLevel.NORMAL,
    status: RequestStatus.OPEN
};
